crates/cxx-qt-build/src/lib.rs

```diff
@@ -186,15 +196,15 @@ fn generate_cxxqt_cpp_files(
     header_dir: impl AsRef<Path>,
 ) -> Vec<GeneratedCppFilePaths> {
     let manifest_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-    let cpp_directory = format!("{}/cxx-qt-gen/src", env::var("OUT_DIR").unwrap());
 
     let mut generated_file_paths: Vec<GeneratedCppFilePaths> = Vec::with_capacity(rs_source.len());
     for rs_path in rs_source {
+        let cpp_directory = format!("{}/cxx-qt-gen/src", env::var("OUT_DIR").unwrap());
```

Comment:
what was the error that meant this needed to be moved?

Reply:
Maybe when changing from reference to value for the others this caused this one to fail as it's used multiple times.

So then is this one better being a reference below so that it can be built once outside the loop ?

Reply:
> what was the error that meant this needed to be moved?

```
   Compiling cxx-qt-build v0.4.1 (/home/przemek/Projects/rust/cxx-qt/crates/cxx-qt-build)
error: the borrowed expression implements the required traits
   --> crates/cxx-qt-build/src/lib.rs:207:71
    |
207 |         generated_file_paths.push(generated_code.write_to_directories(&cpp_directory, &header_dir));
    |                                                                       ^^^^^^^^^^^^^^ help: change this to: `cpp_directory`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
    = note: `-D clippy::needless-borrow` implied by `-D warnings`

error: could not compile `cxx-qt-build` due to previous error
```

but removal of `&` forced me to move the line inside the loop.
`AsRef::<Path>::as_ref(&cpp_directory)` helped. Is that pretty enough?
I guess using only `&` was ambiguous? Or?

Reply:
Hmm I just thought that the following would make more sense, as it constructs the cpp_directory once rather than for every rs_path then hands it a reference. But if it really doesn't work then guess we have to build it in each loop. And i wondered why clippy didn't complain about this before with the old code.

```rust
let cpp_directory = format!("{}/cxx-qt-gen/src", env::var("OUT_DIR").unwrap());

for rs_path in rs_source {
   ...
   generated_file_paths.push(generated_code.write_to_directories(&cpp_directory, &header_dir));
}
```

Reply:
Maybe creating it inside the loop is better and passing by value, i was just more curious what was happening :-)

I guess that the borrow is dereferenced straight away anyway which then causes the copy ? So it thinks it is no worse to have it inside the loop and be by value ?

In that case probably just move it back to doing cpp_directory inside the loop like you had with by-value, sorry.

Reply:
> Hmm I just thought that the following would make more sense, as it constructs the cpp_directory once rather than for every rs_path then hands it a reference. But if it really doesn't work then guess we have to build it in each loop. And i wondered why clippy didn't complain about this before with the old code.

That's also makes more sense to me, as the variable will be the same always. Does Rust compiler optimize that such of things?